src/plugins/expressions/common/expression_renderers/types.ts

```diff
@@ -83,5 +81,10 @@ export interface IInterpreterRenderHandlers {
   update: (params: any) => void;
   event: (event: any) => void;
   getRenderMode: () => RenderMode;
-  uiState?: PersistedState;
+  /**
+   * This uiState interface is actually `PersistedState` from the visualizations plugin,
+   * but expressions cannot know about vis or it creates a mess of circular dependencies.
+   * Downstream consumers of the uiState handler will need to cast for now.
+   */
+  uiState?: unknown;
```

Comment:
We need to get rid of `uiState` in expressions eventually anyway, so this is just another reason to do so 😉